tasks/libs/types.py

```diff
@@ -38,22 +38,30 @@ def key(self):
         return (self.name, self.package)
 
 
-class FailedJobReason(Enum):
+class FailedJobType(Enum):
     JOB_FAILURE = 1
     INFRA_FAILURE = 2
 
 
+class FailedJobReason(Enum):
+    MAIN_RUNNER = 1
+    DOCKER_RUNNER = 2
+    K8S_RUNNER = 3
+    KITCHEN_AZURE = 4
+    FAILED_JOB_SCRIPT = 5
+
+
 class SlackMessage:
     JOBS_SECTION_HEADER = "Failed jobs:"
     INFRA_FAILURES_SECTION_HEADER = "Infrastructure failures:"
     TEST_SECTION_HEADER = "Failed unit tests:"
     MAX_JOBS_PER_TEST = 2
 
-    def __init__(self, base_message, jobs=None):
-        self.base_message = base_message
-        self.failed_jobs = [job for job in jobs if job["failure_type"] == FailedJobReason.JOB_FAILURE] if jobs else []
+    def __init__(self, base="", jobs=None):
+        self.base_message = base
+        self.failed_jobs = [job for job in jobs if job["failure_type"] == FailedJobType.JOB_FAILURE] if jobs else []
```

Comment:
nit: maybe this to simplify the two assignments below?
```suggestion
    def __init__(self, base="", jobs=None):
        jobs = jobs if jobs else []
        self.base_message = base
        self.failed_jobs = [job for job in jobs if job["failure_type"] == FailedJobType.JOB_FAILURE]
```